lib/attrsets.nix

```
@@ -542,6 +542,36 @@ rec {
     attrs:
     map (name: f name attrs.${name}) (attrNames attrs);
 
+  /*
+    Deconstruct an attrset to a list of name-value pairs as expected by [`builtins.listToAttrs`](https://nixos.org/manual/nix/stable/language/builtins.html#builtins-listToAttrs).
+    Each element of the resulting list is an attribute set with these attributes:
+    - `name` (string): The name of the attribute
+    - `value` (any): The value of the attribute
+
+    The following is always true:
+    ```nix
+    builtins.listToAttrs (attrsToList attrs) == attrs
+    ```
+
+    :::{.warning}
+    The opposite is not always true however:
+    ```nix
+    attrsToList (builtins.listToAttrs list) != list
+    ```
```

This reads almost like a double negation because of the `:`.

```suggestion
    The opposite is not always true however. In general:
    ```nix
    attrsToList (builtins.listToAttrs list) != list
    ```
```